### PR TITLE
Add Coordinates Module and Tests

### LINKED ISSUE #115: BL and Normal Coordinate Transformation : Confusion in Names!
🐞 **Problem**

There is lot of confusion in coordinates transformation! 
`bl_coord_transforms` and `coord_transforms` are not good names as in the `coord_transforms` does not specify the type of coordinates it works on while `bl_coord_transforms` clearly tells user what the module does! There is a need to make the coord transforms more intuitive.

🎯 **Goal**

Make coordinate transformations more intuitive in EinsteinPy

💡 **Possible solutions**

* Rename `coord_transforms.py`
* Merge both files into one
* Think of some other solution?

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #115`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### LINKED ISSUE #121: Clean the Messed up utils!
🐞 **Problem**

The `utils` module is currently too messed up. 

🎯 **Goal**

💡 **Possible solutions**

Make necessary changes to clean it. Refactor modules, code.

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### PR DESCRIPTION
**Pull request by rishi-s8**
Partially Fixes #121 
Completely fixes #115 #118 

@shreyasbapat

### CODE REVIEW

File: src/einsteinpy/coordinates/__init__.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from einsteinpy.coordinates.core import BoyerLindquist, Cartesian, Spherical
+
+from einsteinpy.coordinates.velocity import (  # isort:skip
```
Comment (shreyasbapat) [inline src/einsteinpy/coordinates/__init__.py]: Can you add a TODO here? Referencing the issue. And create an issue in EInsteinPy too! With link to the issue in isort

File: src/einsteinpy/coordinates/_base.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import astropy.units as u
```
Comment (shreyasbapat) [inline src/einsteinpy/coordinates/_base.py]: Remove this file once in for all!

File: src/einsteinpy/coordinates/velocity.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+class CartesianDifferential(Cartesian):
+    """
+    Class for calculating and transforming the velocity
+    """
+
+    @u.quantity_input(
+        x=u.km, y=u.km, z=u.km, v_x=u.km / u.s, v_y=u.km / u.s, v_z=u.km / u.s
+    )
+    def __init__(self, x, y, z, v_x, v_y, v_z):
+        super(CartesianDifferential, self).__init__(x, y, z)
```
Comment (shreyasbapat) [inline src/einsteinpy/coordinates/velocity.py]: Can you look at the docstrings in `core.py` and add similar docstrings here?

File: src/einsteinpy/coordinates/velocity.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        return SphericalDifferential(
+            transformed_spherical.r,
+            transformed_spherical.theta,
+            transformed_spherical.phi,
+            v_r,
+            v_t,
+            v_p,
+        )
+
+    def boyerlindquist_differential(self, a):
```
Comment (shreyasbapat) [inline src/einsteinpy/coordinates/velocity.py]: Rename this to `bl_differential`

Comment (shreyasbapat) [reply src/einsteinpy/coordinates/velocity.py]: Also, a should be with units :)

Comment (shreyasbapat) [reply src/einsteinpy/coordinates/velocity.py]: https://github.com/einsteinpy/einsteinpy/pull/176/files#diff-517ed4ee5def3bce49f42958bec83fc4R207
See this

File: src/einsteinpy/coordinates/velocity.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        return CartesianDifferential(
+            transformed_cartesian.x,
+            transformed_cartesian.y,
+            transformed_cartesian.z,
+            v_x,
+            v_y,
+            v_z,
+        )
+
+    def boyerlindquist_differential(self, a):
```
Comment (shreyasbapat) [inline src/einsteinpy/coordinates/velocity.py]: Again bl_differential, and units to a

File: src/einsteinpy/coordinates/velocity.py [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    @u.quantity_input(
+        r=u.km, theta=u.rad, phi=u.rad, v_r=u.km / u.s, v_t=u.rad / u.s, v_p=u.rad / u.s
+    )
+    def __init__(self, r, theta, phi, v_r, v_t, v_p):
+        super(BoyerLindquistDifferential, self).__init__(r, theta, phi)
+        self.v_r = v_r
+        self.v_t = v_t
+        self.v_p = v_p
+
+    def cartesian_differential(self, a):
```
Comment (shreyasbapat) [inline src/einsteinpy/coordinates/velocity.py]: Units to a ! Plus doctrings everywhere!

**Comment by rishi-s8** [comment]
Thanks for reviewing. I will do these changes :)

File: src/einsteinpy/coordinates/__init__.py (hunk 1 above)
Comment (rishi-s8) [reply src/einsteinpy/coordinates/__init__.py]: ```TODO``` keyword was causing CodeFactor to fail. Added a descriptive comment instead.

**Comment by shreyasbapat** [comment]
Thanks a lot for the efforts! :)

**Comment by shreyasbapat** [comment]
With your permission, I added an extra commit in your PR

**Comment by shreyasbapat** [comment]
Thanks @rishi-s8 :)

**Comment by rishi-s8** [comment]
Happy to do this.. :)

**Pull request merged**